File: novel-library-service/src/main/java/de/roamingthings/tracing/novellibrary/configuration/ClockConfiguration.java
package de.roamingthings.tracing.novellibrary.configuration;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.time.Clock;

import static java.time.Clock.systemDefaultZone;

@Configuration
public class ClockConfiguration {
    @Bean
    public Clock systemClock() {
        return systemDefaultZone();
    }
}

File: novel-ai-service/src/main/kotlin/de/roamingthings/tracing/novelai/configuration/ClockConfiguration.kt
package de.roamingthings.tracing.novelai.configuration

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

import java.time.Clock

import java.time.Clock.systemDefaultZone

@Configuration
class ClockConfiguration {
    @Bean
    fun systemClock(): Clock {
        return systemDefaultZone()
    }
}
